dash/examples/bench.13.allreduce/main.cpp

```diff
@@ -153,10 +181,56 @@ measurement evaluate(int reps, std::string testcase, benchmark_params params)
       g_max.get().fetch_op(dash::max<value_t>(), std::numeric_limits<value_t>::max());
 
       team.barrier();
+    } else if (testcase == "dart_allreduce.custom") {
+      minmax_t min_max_in{lmin, lmax};
+      minmax_t min_max_out{};
+      dart_datatype_t  new_type;
+      dart_operation_t new_op;
+      dart_type_create_custom(sizeof(minmax_t), &new_type);
+      dart_op_create(
+        &minmax_fn, NULL, true, new_type, true, &new_op);
+      dart_allreduce(
+          &min_max_in,                        // send buffer
+          &min_max_out,                       // receive buffer
+          1,                                  // buffer size
+          new_type,                           // data type
+          new_op,                             // operation
+          dash::Team::All().dart_id()         // team
+          );
+      dart_type_destroy(&new_type);
+      dart_op_destroy(&new_op);
+    } else if (testcase == "dart_allreduce.lambda") {
+      minmax_t min_max_in{lmin, lmax};
+      minmax_t min_max_out{};
+      dart_datatype_t  new_type;
+      dart_operation_t new_op;
+      dart_type_create_custom(sizeof(minmax_t), &new_type);
+      auto fn = [](const minmax_t &a, const minmax_t &b){
+        minmax_t res = b;
+        if (a.min < res.min) {
+          res.min = a.min;
+        }
+        if (a.max > res.max) {
+          res.max = a.max;
+        }
+        return res;
+      };
+      dart_op_create(
+        &minmax_lambda<decltype(fn)>, &fn, true, new_type, true, &new_op);
```

Comment:
can you please document the parameters here, especially the boolean values. Either as a comment or, even better, in the form of variable names.

Reply:
Done.